apps/console/src/features/saml2-configuration/pages/saml2-configuration.tsx

```diff
@@ -0,0 +1,392 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentityAppsApiException } from "@wso2is/core/exceptions";
+import { hasRequiredScopes } from "@wso2is/core/helpers";
+import { AlertLevels, IdentifiableComponentInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { URLUtils } from "@wso2is/core/utils";
+import { Field, Form, FormPropsInterface } from "@wso2is/form";
+import { EmphasizedSegment, PageLayout, PrimaryButton, URLInput } from "@wso2is/react-components";
+import { FormValidation } from "@wso2is/validation";
+import React, { FunctionComponent, MutableRefObject, ReactElement, useEffect, useMemo, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { Dispatch } from "redux";
+import { Divider, Grid, Placeholder, Ref } from "semantic-ui-react";
+import { AppConstants, AppState, FeatureConfigInterface, history } from "../../core";
+import { updateSaml2Configurations, useSaml2Config } from "../api/saml2-configuration";
+import { Saml2ConfigurationConstants } from "../constants/saml2-configuration";
+import {
+    Saml2ConfigAPIResponseInterface,
+    Saml2ConfigFormErrorValidationsInterface,
+    Saml2ConfigFormValuesInterface
+} from "../models/saml2-configuration";
+
+/**
+ * Props for saml2 configuration settings page.
+ */
+type Saml2ConfigurationPageInterface = IdentifiableComponentInterface;
+
+const FORM_ID: string = "saml2-config-form";
+
+/**
+ * saml2 configuration page.
+ */
+export const Saml2ConfigurationPage: FunctionComponent<Saml2ConfigurationPageInterface> = (
+    props: Saml2ConfigurationPageInterface
+): ReactElement => {
+    const { [ "data-componentid" ]: componentId } = props;
+
+    const pageContextRef: MutableRefObject<any> = useRef(null);
+    const formRef: MutableRefObject<FormPropsInterface> = useRef<FormPropsInterface>(null);
+    const url: MutableRefObject<HTMLDivElement> = useRef<HTMLDivElement>();
+
+    const featureConfig : FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+    const allowedScopes : string = useSelector((state: AppState) => state?.auth?.allowedScopes);
+    
+    const isReadOnly : boolean = useMemo(() => !hasRequiredScopes(
+        featureConfig?.residentIdp,
+        featureConfig?.residentIdp?.scopes?.update,
+        allowedScopes
+    ), [ featureConfig, allowedScopes ]);
+
+    const dispatch : Dispatch<any> = useDispatch();
+
+    const { t } = useTranslation();
+
+    const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+    const [ saml2Config , setSaml2Config ] = 
+        useState<Saml2ConfigFormValuesInterface>(undefined);
+    const [ destinationUrls, setDestinationUrls ] = useState("");
+    const [ showURLError, setShowURLError ] = useState(false);
```

Comment:
Shall we add the types?

Reply:
Fixed with 1d3ef8fd4f615de27060c959ea91a56ca1f3158f